selfservice/hook/web_hook.go

```diff
@@ -332,40 +344,30 @@ func (e *WebHook) execute(ctx context.Context, data *templateContext) error {
 			if canInterrupt || parseResponse {
 				if err := parseWebhookResponse(resp, data.Identity); err != nil {
 					span.SetStatus(codes.Error, err.Error())
-					errChan <- err
+					return err
 				}
 			}
-			errChan <- fmt.Errorf("webhook failed with status code %v", resp.StatusCode)
-			return
+			return fmt.Errorf("webhook failed with status code %v", resp.StatusCode)
 		}
 
 		if parseResponse {
 			if err := parseWebhookResponse(resp, data.Identity); err != nil {
 				span.SetStatus(codes.Error, err.Error())
-				errChan <- err
+				return err
 			}
 		}
 
-		errChan <- nil
-	}()
-
-	if ignoreResponse {
-		traceID, spanID := span.SpanContext().TraceID(), span.SpanContext().SpanID()
-		logger := e.deps.Logger().WithField("otel", map[string]string{
-			"trace_id": traceID.String(),
-			"span_id":  spanID.String(),
-		})
-		go func() {
-			if err := <-errChan; err != nil {
-				logger.WithField("duration", time.Since(startTime)).WithError(err).Warning("Webhook request failed but the error was ignored because the configuration indicated that the upstream response should be ignored.")
-			} else {
-				logger.WithField("duration", time.Since(startTime)).Info("Webhook request succeeded")
-			}
-		}()
 		return nil
 	}
 
-	return <-errChan
+	if !ignoreResponse {
+		return makeRequest()
+	}
+	go func() {
+		// we cannot handle the error as we are running async, and it is logged anyway
+		_ = makeRequest()
+	}()
+	return nil
```

Comment:
```suggestion
	if ignoreResponse {
		go func() {
		    // we cannot handle the error as we are running async, and it is logged anyway
		    _ = makeRequest()
	    }()
	    return nil
	}
	return makeRequest()
```

Just a nit. I like positive branching. But not sure that it improves this much. It's fine either way.

Reply:
Hm I typically go for shorter branches in the if body, as that makes it easier to read.